plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/persistence/LibraryClasspathContainerSerializer.java

```diff
@@ -0,0 +1,109 @@
+/*******************************************************************************
+ * Copyright 2016 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+package com.google.cloud.tools.eclipse.appengine.libraries.persistence;
+
+import com.google.cloud.tools.eclipse.appengine.libraries.LibraryClasspathContainer;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.logging.Logger;
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.IFolder;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.jdt.core.IJavaProject;
+import org.osgi.framework.FrameworkUtil;
+
+public class LibraryClasspathContainerSerializer {
+
+  private static final Logger logger = Logger.getLogger(LibraryClasspathContainerSerializer.class.getName());
+
+  public interface LibraryContainerStateLocationProvider {
+    IPath getContainerStateFile(IJavaProject javaProject, IPath containerPath, boolean create) throws CoreException;
+  }
+
+  private LibraryContainerStateLocationProvider stateLocationProvider;
+  
+  public LibraryClasspathContainerSerializer() {
+    this(new DefaultStateLocationProvider());
+  }
+  
+  @VisibleForTesting
+  public LibraryClasspathContainerSerializer(LibraryContainerStateLocationProvider stateLocationProvider) {
+    this.stateLocationProvider = stateLocationProvider;
+  }
+
+  public void saveContainer(IJavaProject javaProject, LibraryClasspathContainer container) throws IOException,
+                                                                                                  CoreException {
+    File stateFile = getContainerStateFile(javaProject, container.getPath(), true);
+    if (stateFile == null) {
+      logger.warning("Container state file cannot be created, save failed");
+      return;
+    }
+    try (PrintWriter outputStream = new PrintWriter(stateFile)) {
```

Comment:
don't use PrintWriter because it swallows exceptions and may not properly internationalize. Try OutputStreamWriter instead